atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -450,6 +456,40 @@ def upload_file(self, session, output, serverdir):
         self.log.debug("uploaded %r", path)
         return path
 
+    def get_output_metadata(self, path, filename):
+        """
+        Describe a file by its metadata.
+
+        :return: dict
+        """
+
+        checksums = get_checksums(path, ['md5'])
+        metadata = {'filename': filename,
+                    'filesize': os.path.getsize(path),
+                    'checksum': checksums['md5sum'],
+                    'checksum_type': 'md5'}
+
+        return metadata
+
+    def upload_scratch_metadata(self, koji_metadata, koji_upload_dir, koji_session):
+        metadata_file = NamedTemporaryFile(prefix="metadata", suffix=".json", mode='wb')
+        metadata_file.write(json.dumps(koji_metadata))
+        metadata_file.flush()
+
+        filename = "metadata.json"
+        meta_output = Output(file=metadata_file,
+                             metadata=self.get_output_metadata(metadata_file.name, filename))
+
+        if meta_output.file:
+            try:
+                self.upload_file(koji_session, meta_output, koji_upload_dir)
+                path = os.path.join(koji_upload_dir, filename)
+                self.log = logging.LoggerAdapter(self.log, {'arch': METADATA_TAG})
+                self.log.info(path)
+                self.log = logging.LoggerAdapter(self.log, {})
```

Comment:
Wonder if resetting the LoggerAdapter should move into the `finally` clause.

Reply:
yeah good idea